### ISSUE TITLE
live-command doesn't work with unranged :Subvert

### ISSUE THREAD

**Issue by gregorias** [smjonas/live-command.nvim#24]
[tpope/vim-abolish](https://github.com/tpope/vim-abolish) provides a useful :Subvert command that does smart substitution.

I've configured it with live-command as so:

```lua
use {
  "smjonas/live-command.nvim",
  config = function()
    require("live-command").setup {
      commands = {
        LSubvert = { cmd = "Subvert" },
      },
    }
  end,
}
```

but I'm getting an error when I try `:LSubvert/send_wire/SendWire/`:

```
Error executing Lua callback: ...packer/start/live-command.nvim/lua/live-command/init.lua:281: Vim(substitute):E486: Pattern not found: \v\C%(send_wire|SEND_WIRE|SendWire)   
stack traceback:                                                                                                                                                              
        [C]: in function 'cmd'                                                                                                                                                
        ...packer/start/live-command.nvim/lua/live-command/init.lua:281: in function 'execute_command'                                                                        
        ...packer/start/live-command.nvim/lua/live-command/init.lua:295: in function <...packer/start/live-command.nvim/lua/live-command/init.lua:288>
```

The plugin works when I provide a range though, e.g., `:%LSubvert/send_wire/SendWire/`.

I can provide a Docker container for reproducing this issue if needed.

**Comment by smjonas** [comment]
Thanks for reporting! Which Neovim version are you using?

Actually running `:%LSubvert/send_wire/SendWire/` crashes Neovim for me with the following error (no idea what's causing this):
```
[...]/neovim/src/nvim/extmark.c:256: extmark_clear: Assertion 'mark.ns > 0 && mark.id > 0' failed
```

What seems to be causing your issue is that `live_command` is executing the command `Subvert /send_wire/send/` (notice the extra space after `Subvert`). Can you try changing your command to `LSubvert = { cmd = "Subvert/" }`? In this case you don't need to write the `/` before `send_wire`.

**Comment by gregorias** [comment]
I use NVIM v0.8.1.

Changing to `LSubvert = { cmd = "Subvert/" }` doesn't help. With that I also can't run queries without a space at the beginning, e.g., `%LSubverthello/world/` doesn't work, but ``%LSubvert hello/world/` does but matches ` hello` not just `hello`.

**Comment by nfrid** [comment]
`:Subvert /foo/bar/` works the same as `:Subvert/foo/bar/` so it's not the cause...

**Comment by smjonas** [comment]
> `:Subvert /foo/bar/` works the same as `:Subvert/foo/bar/` so it's not the cause...

You are right actually, my bad... Does running the command `:LSubvert/foo/bar/` make your Neovim crash as well?

**Comment by gregorias** [comment]
> > `:Subvert /foo/bar/` works the same as `:Subvert/foo/bar/` so it's not the cause...
> 
> You are right actually, my bad... Does running the command `:LSubvert/foo/bar/` make your Neovim crash as well?

No, I'm only getting an error message.

```
Error executing Lua callback: ...k/packer/opt/live-command.nvim/lua/live-command/init.lua:281: Vim(substitute):E486: Pattern not found: \v\C%(Foo|foo|FOO)                                                                      
stack traceback:                                                                                                                                                                                                                
        [C]: in function 'cmd'                                                                                                                                                                                                  
        ...k/packer/opt/live-command.nvim/lua/live-command/init.lua:281: in function 'execute_command'                                                                                                                          
        ...k/packer/opt/live-command.nvim/lua/live-command/init.lua:295: in function <...k/packer/opt/live-command.nvim/lua/live-command/init.lua:288>
```

**Comment by nfrid** [comment]
> You are right actually, my bad... Does running the command `:LSubvert/foo/bar/` make your Neovim crash as well?

Yes, with the same `Assertion 'mark.ns > 0 && mark.id > 0' failed` message, if I use % or range, just as I start to type `bar`
And it **segfaults** on inline (i.e. no % or range) as I start to type `foo`.

I run neovim nightly though. I built a fresh git version yesterday.
I tried to install and run neovim 0.8.1 from Arch repos. It works the same...

**Comment by smjonas** [comment]
Ok I've done some debugging regarding the segfault and here's what I found out:

TL;DR: The builtin LSP might be the cause of the crash... :thinking: Please help me confirm or disprove this (see last section below).

## Minimal reproduction
The error is not directly related to `vim-abolish` or `live-command.nvim` but seems to be a Neovim bug as shown by the following reproduction steps. 

1. Paste the following code in your Lua config and run `:so %` to source it:
```lua
local old = 1
vim.api.nvim_create_user_command("Crash", function()
  vim.cmd("s/old/new/")
end, {
  preview = function()
    vim.cmd("s/old/new/")
    return 2
  end,
})
```
2. Open any file that has an LSP server attached to it and run the `:Crash` command. You might have to run the command twice. Wait for a short while, then Neovim should crash.

## Please help with confirming the cause of this crash
@nfrid: did you run the command while an LSP server was running by any chance? 

@gregorias can you check if you can reproduce the crash (not just the error) when running `:LSubvert/foo/bar/` with a buffer containing `foo` or following the above reproduction steps?

**Comment by nfrid** [comment]
> @nfrid: did you run the command while an LSP server was running by any chance?

nope, I ran it w/out lsp as well, same result:
![image](https://user-images.githubusercontent.com/13046595/208305088-1689c2dd-3917-493c-9668-0cb69d85c6dd.png)
![image](https://user-images.githubusercontent.com/13046595/208305114-eb502509-b576-4959-997b-2d04613546f8.png)
![image](https://user-images.githubusercontent.com/13046595/208305179-ab2f861e-6786-43af-8fd7-1ec2c348b9e3.png)

**Comment by nfrid** [comment]
also, the crash happens immediately after I type second `/`

**Comment by smjonas** [comment]
Ok thank you, I was able to reproduce the crash with your reproduction steps. That seem to be two different issues then.

**Comment by smjonas** [comment]
I created https://github.com/neovim/neovim/issues/21495

**Comment by smjonas** [comment]
Resolved by https://github.com/neovim/neovim/pull/23809
